src/components/presentational/DataChart/DataChart.tsx

```diff
@@ -0,0 +1,242 @@
+import React, { useContext } from 'react'
+import { add, format, isToday } from 'date-fns'
+import { CardTitle, LayoutContext, LoadingIndicator } from '../../presentational'
+import { Area, AreaChart, Bar, BarChart, CartesianGrid, Cell, Line, LineChart, ReferenceLine, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'
+import "./DataChart.css"
+import { AxisDomain } from 'recharts/types/util/types'
+import { ColorDefinition, resolveColor } from '../../../helpers/colors'
+
+export interface DataChartProps {
+    title?: string
+    intervalType?: "Week" | "Month",
+    intervalStart: Date,
+    data: any[],
+    dataKeys?: string[],
+    hasAnyData: boolean,
```

Comment:
I'm not sure there's much utility in moving hasAnyData and hideIfNoData down to this; feels like it's actually putting in more code than you're saving?

Reply:
More code in what way?  This seemed in line with the paradigm of the parent container doing the "heavy lifting" for the data and then passing parameters down to the child component to tell it how it needs to render at any point in time.  The empty graph placeholder is useful - so if these properties aren't passed down here, then the block of code for displaying the "No Data" label or the `LoadingIndicator` needs to get copy/pasted in both the `DailyDataChart` and the `SurveyDataChart`, right?  That's what I was avoiding as it explicitly felt like "more code."

As Renee noted above, I didn't really deal well with these for the SurveyDataChart; I'd punted on them in my initial push of doing this work a few months ago, and then I forgot that I didn't deal very well with them.  That was mainly because it didn't feel like there was a good equivalent for the `checkAvailability` call - but as noted above, I wanted to leverage the empty graph with the loading indicator.

Reply:
Ah I forgot about the "noData" label.  Nevermind - I thought this was just to check for hiding the graph